graphql/schema.graphql

```diff
@@ -157,6 +157,74 @@ type Query {
 
+  sponsoredProducts(
+    """
+    Search term. It can contain any character.
+    """
+    fullText: String = ""
+    """
+    List of the selected facets
+
+    The order in which the terms appear is not relevant to the search.
+
+    You can also repeat the same `facetKey` several times for different values.
+    """
+    selectedFacets: [SelectedFacetInput]
+    """
+    Filter by price range. e.g.: {a} TO {b} - {a} is the minimum price "from" and {b} is the highest price "to"
+    """
+    priceRange: String = ""
+    """
+    Filter by availability at a specific sales channel. e.g.: salesChannel:4 if want filter by available products for the sales channel 4
+    """
+    salesChannel: String = ""
+    """
+    Order by a criteria. OrderByPriceDESC/OrderByPriceASC, OrderByTopSaleDESC, OrderByReviewRateDESC, OrderByNameASC/OrderByNameDESC, OrderByReleaseDateDESC, OrderByBestDiscountDESC, OrderByScoreDESC
+    If you want to sort by a specification, use the format {specification key}:{asc|desc}. For example: "pricePerUnit:asc" or "size:desc" (this only works on `vtex.search-resolver@1.x`)
+    """
+    orderBy: String = "OrderByScoreDESC"
+    """
+    Pagination item start
+    """
+    from: Int = 0
+    """
+    Pagination item end
+    """
+    to: Int = 9
+    """
+    If true, uses isAvailablePerSalesChannel_ parameter on query with segment's sales channel. Will override any given salesChannel arg
+    """
+    hideUnavailableItems: Boolean = false
+    """
+    If you want faster searches and do not care about most up to date prices and promotions, use skip value.
+    """
+    simulationBehavior: SimulationBehavior = default
+    """
+    Each search engine has its own database, but this database might not have all the product information like `clusterHighlights` or `productClusters`.
+    As an alternative, the search engine may use the VTEX API to complete this information by setting this field to true.
+    """
+    productOriginVtex: Boolean = false
+    """
+    Indicates how the search-engine will deal with the fullText if there is more than one word. Set `and` if the returned products must have all the words in its metadata or `or` otherwise.
+    """
+    operator: Operator
+    """
+    Indicates how the search engine will correct misspelled words by using fuzzy logic.
+    It can be a number representing the max number of misspelled letters, or the string `auto` suggesting that the search-engine should set this value by itself.
+    """
+    fuzzy: String
+    """
+    As fuzzy and operator, it controls the search state, but it is for general purposes. This field allows the search engines to apply features that are not handled by the other fields.
+    The possible values in this field are defined by each search engine.
+    """
+    searchState: String
+
+    """
+    Search options that customize the search result.
+    """
+    options: Options
+  ): [Product] @cacheControl(scope: SEGMENT, maxAge: SHORT) @withSegment
```

Comment:
@hiagolcm Hey, I think we need a few extra info other than the product itself. I'd suggest this type:

```graphql
type SponsoredProduct {
  """
  Unique identifier of the Ad Item.
  """
  adId: ID

  """
  Campaign ID that's associated with this ad.
  """  
  campaignId: ID

  """
  ID of the request for this ad.
  """
  adRequestId: ID
  
  """
  ID of the response for this ad.
  """
  adResponseId: ID

  """
  Potential cost of a click in this product.
  """
  costPerClick: Float

  """
  The sponsored product.
  """
  product: Product
}
```

I'll also comment that in the `adserver-graphql`.

Reply:
As we already discussed this version will only have the fields that are necessary for the relevance to work. when you finish the development we can add the new fields
thanks